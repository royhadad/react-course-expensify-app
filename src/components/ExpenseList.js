import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="expense-list">
            <div className="expense-list-header">
                <div className="show-for-mobile">Expenses</div>
                <div className="show-for-desktop">Expenses</div>
                <div className="show-for-desktop">Amount</div>
            </div>
            <div className="list-body">
            {
                props.expenses.length === 0 ? (
                    <span className="list-item--message">No expenses</span>
                ) : (
                        props.expenses.map((expense) => {
                            return <ExpenseListItem key={expense.id} {...expense} />;
                        })
                    )
            }
            </div>
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);
