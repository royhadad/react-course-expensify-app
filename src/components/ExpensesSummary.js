import React from 'react';
import { connect } from 'react-redux';
import getExpensesTotal from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses';
import numeral from 'numeral';

export class ExpensesSummary extends React.Component {
    render() {
        return (
            <div>
                <p>
                    viewing {this.props.expenseCount} expense{this.props.expenseCount === 1 || 's'} totalling {numeral(this.props.expensesTotal / 100).format('$0,0.00')}
                </p>
            </div>
        );
    }
}
const mapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: getExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);