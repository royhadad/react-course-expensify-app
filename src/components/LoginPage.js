import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export class LoginPage extends React.Component {
    render() {
        return (
            <div>
                <h1>Login</h1>
                <button onClick={this.props.startLogin}>Login</button>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
