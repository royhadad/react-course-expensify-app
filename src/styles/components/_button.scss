.button {
    font-size: $font-size-large;
    background: $blue;
    padding: $s-size;
    border: none;
    font-weight: 300;
    color: white;
    display: inline-block;
    text-decoration: none;
    line-height: 1;
}

.button--link {
    @extend .button;
    background: none;
}

.button--secondary {
    @extend .button;
    background: $light-gray;
}