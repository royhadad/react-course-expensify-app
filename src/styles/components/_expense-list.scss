.expense-list {
}

.expense-list-header {
    background: $off-white;
    display: flex;
    justify-content: space-between;
    border:  1px solid $border-gray;
    > div {
        padding: $s-size $m-size;
        color: $gray;
    }
}

.list-body {
    margin-bottom: $m-size;
    @media (min-width: $desktop-breakpoint) {
        margin-bottom: $l-size;
    }
}

.list-item {
    display: flex;
    flex-direction: column;
    border:  1px solid $border-gray;
    border-top: none;
    color: $dark-gray;
    padding: $s-size;
    text-decoration: none;
    transition: background .3s ease;
    &:hover {
        background: $off-white;
    }
    @media (min-width: $desktop-breakpoint) {
        align-items: center;
        flex-direction: row;
        justify-content: space-between;
        padding: $m-size;
    }
}

.list-item--message {
    @extend .list-item;
    align-items: center;
    justify-content: center;
    color: $gray;
    padding: $m-size;
    &:hover {
        background: none;
    }
}

.list-item__title {
    margin: 0;
    word-break: break-all;
}

.list-item__sub-title {
    color: $gray;
    font-size: $font-size-small;
}

.list-item__data {
    margin: $s-size 0 0 0;
    @media (min-width: $desktop-breakpoint) {
        margin: 0;
        padding-left: $s-size;
    }
}