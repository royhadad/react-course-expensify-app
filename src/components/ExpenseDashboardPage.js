import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import getExpensesTotal from '../selectors/expenses-total';

const ExpenseDashboardPage = () => (
  <div>
    {getExpensesTotal([{amount:4}])}
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
