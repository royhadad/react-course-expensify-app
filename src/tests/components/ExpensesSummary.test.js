import React from 'react';
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should match snapshot with all 3 expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={expenses.length} expensesTotal={114195} />);
    expect(wrapper).toMatchSnapshot();
});

test('should match snapshot with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={expenses[0].amount} />);
    expect(wrapper).toMatchSnapshot();
});