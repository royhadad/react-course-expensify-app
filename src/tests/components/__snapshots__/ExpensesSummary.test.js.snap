// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with 1 expense 1`] = `
<div>
  <p>
    viewing 
    1
     expense
    <Component />
     totalling 
    $1.95
  </p>
</div>
`;

exports[`should match snapshot with all 3 expenses 1`] = `
<div>
  <p>
    viewing 
    3
     expense
    s
     totalling 
    $1,141.95
  </p>
</div>
`;
